### DISCUSSION TITLE
ROI's

### DISCUSSION THREAD

**Discussion by springbok** [springbok/MLM2PRO-GSPro-Connector#19] [General]
Just thinking about this, with the R option I could present a menu that would allow you to select:

```
A - All
1 - Ball Speed
2 - ....
```

so people can select a specific roi to update or do all. 

On error of an roi I could popup the roi editor for the specific roi that caused the error but not sure that is overkill maybe just allow the user to select the one they wish to update then resume.

Thought's?

**Comment by wonder99** [comment]
I've been thinking and I'm not sure the best direction here.
When I get a bad reading, I find it convenient to close the connector, edit
the one corner of the one metric that failed in the JSON, and restart the
connector.  I don't really want to re-pick the other metrics or even the
other corners of the bad one.  That's because I can normally tell which
direction it needs to be tweaked.  Stripping the trailing dots helps with
this, which may just leave the need to widen the box.  That's what I find
is the fix, most of the time.
We could consider an automatic hunting approach, where the connector tries
to identify the direction to move the boundary.  That may be overkill, I
realize.  Failing that, I suppose the menu you proposed is pretty good.
Also, we should recommend people maximize the screen before clicking the
boundaries, for best accuracy.

What might be very handy is a graphical view of what was captured, with an
overlay of the bounding box--to clearly show the user where the problem
was, so when he picks it next time, it'll be right.

Also, for me and a few others who run the MLM app on various devices, I
really like the feature I added to the other connector which optionally
lets you specify the json file name on the command line.  Let's me switch
devices easily.  I really don't want to pick bounding boxes any more often
than absolutely necessary.

- Paul

**Reply by springbok** [reply]
I agree the GUI would be the ultimate goal so it's clear where the bounding boxes are, how they line up, and allow tuning if required. I will have a look and see if its possible to implement something like this as it is a bit confusing at the moment especially when you are new to the connector.

Catering for various devices is a great idea, one solution would be for the user to create a rois file for each device and name it accordingly, e.g. iphone_rois.json, ipad_rois.jason, etc. At startup I could search for all files with a file name with the format [device name]_rois.json and present it as a selection menu, the user can then pick which device to use for that session.

```
1. iphone
2. ipad
3. Android
4. ...
```
These files would also store the window dimensions as well as the roi's.

**Comment by wonder99** [comment]
Another thing I added to my local copy--a sound notification when the OCR
had a misread.  I have different sounds for 'all dashes', and one or more
bad values.  I find that handy because the connector window is hidden while
you're playing.

**Reply by springbok** [reply]
Do you reckon it's worth adding it into the connector or is it only useful for debugging purposes?
